platform/core/src/poly_out.rs

```diff
@@ -0,0 +1,112 @@
+use std::fs::DirBuilder;
+use std::path::Path;
+use uuid::Uuid;
+
+type ResourceId = String;
+type FsId = String;
+
+struct Metadata {}
+struct Content {}
+
+trait PolyOutTrait {
+    fn metadata(resource_id: ResourceId) -> Result<Metadata, String>;
+    fn read(resource_id: ResourceId) -> Result<Content, String>;
```

Comment:
What would this do for a directory? Probably worth documenting.

Reply:
It will return the files and folders contained by the folder, one level deep. Still thinking how, but probably as a string.